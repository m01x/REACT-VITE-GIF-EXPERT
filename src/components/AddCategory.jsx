import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = (evento) => {
    setInputValue(evento.target.value);
  };

  const onSubmitForm = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return; //no hay nada, return!

    //Yo lo hice de otra manera mas corta, fernando opto por el camino largo. Se anota igual.
    //setCategories(categories => [inputValue, ...categories]);

    setInputValue('');
    onNewCategory(inputValue.trim());
  };

  return (
    <form action="" onSubmit={onSubmitForm} aria-label='form'>
      {' '}
      {/*Recordar que esto es lo mismo que onSubmit = {(event)=> onSubmit(event)} */}
      <input
        type="text"
        placeholder="Buscar Gif ðŸ”Ž"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}