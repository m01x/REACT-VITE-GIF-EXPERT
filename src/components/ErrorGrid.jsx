

export const ErrorGrid = () => {
    return (
        <div className="card">
            <img src="https://media4.giphy.com/media/YyKPbc5OOTSQE/giphy.gif?cid=fd6452dazyyojqm4soolucq2ogpecgvgp2ml7mo40o8bgb7n&amp;ep=v1_gifs_search&amp;rid=giphy.gif&amp;ct=g" alt="art error GIF by kidmograph"></img>
            <p>No encontramos nada capitan!</p>
        </div>
    )
}
