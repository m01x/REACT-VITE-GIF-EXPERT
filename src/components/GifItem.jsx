import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {

    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}


/**
 * TODO: Hacer la tarea , a√±adir PropTypes, (yarn add), title obligatorio, url obligatorioa... evaluar el snapshot.
 */

GifItem.propTypes = {
    title: PropTypes.string.isRequired, // title debe ser una cadena
    url: PropTypes.string.isRequired    // url debe ser una cadena (no existe PropTypes.url)
}