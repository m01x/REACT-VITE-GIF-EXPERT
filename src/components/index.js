//Archivo de barril, para evitar importar todo, uno por uno.... importas esto y ya esta.

export * from './AddCategory'
export * from './GifGrid'
export * from './GifItem'
export * from './ErrorGrid'