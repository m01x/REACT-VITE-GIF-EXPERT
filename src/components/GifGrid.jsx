import { useEffect, useState } from "react"
import { GifItem } from "./GifItem";
import { getGifs } from "../helpers/getGifs"

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    const getImages = async () => {
        const newImages = await getGifs(category);
        setImages(newImages)
    }

    useEffect(() => {
        getImages();

    }, [])


    return (
        <div key={category}>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem
                            key={image.id}
                            //omg!! wea wena!
                            {...image}
                        />
                    ))
                }
            </div>
        </div >
    )
}
