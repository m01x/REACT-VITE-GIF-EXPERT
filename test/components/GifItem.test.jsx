import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

/**
 * *Tuve que recurrir a reparar el algunos archivos Jest.config, nada del otro mundo.
 */

describe('Pruebas en <GifItem/>', () => { 
    
    const title = 'makima';
    const url = 'https://media0.giphy.com/media/VIcRzb0tOPwk7ebiqM/100.gif?cid=fd6452dac4aju8daynrbfewi149ph14auj1gu7d4cp93wk0h&ep=v1_gifs_search&rid=100.gif&ct=g';

    test('Test - Match con el Snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        
        expect(container).toMatchSnapshot();

    });


    test('Debe mostrar la imagen con el url y el ALT indicado', () => { 
        
        render( <GifItem title={title} url={url}/> );
        //Para ver lo renderizado, screen.debug();

        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe( url );
        expect(alt).toBe( alt );
     });


     test('debe Mostrar el titulo en el componente', () => {
        render( <GifItem title={title} url={url}/> );
        expect( screen.getByText( title )).toBeTruthy();

     });



});