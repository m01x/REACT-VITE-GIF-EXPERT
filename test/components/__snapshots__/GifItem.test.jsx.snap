// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Test - Match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="makima"
      src="https://media0.giphy.com/media/VIcRzb0tOPwk7ebiqM/100.gif?cid=fd6452dac4aju8daynrbfewi149ph14auj1gu7d4cp93wk0h&ep=v1_gifs_search&rid=100.gif&ct=g"
    />
    <p>
      makima
    </p>
  </div>
</div>
`;
